<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>707</width>
    <height>454</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Modbus Master</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QLabel" name="lblSettings">
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Settings</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayoutSettings">
      <item row="1" column="4">
       <widget class="QLabel" name="lblFormat">
        <property name="text">
         <string>Format</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="lblModbusMode">
        <property name="text">
         <string>Modbus Mode</string>
        </property>
        <property name="buddy">
         <cstring>cmbModbusMode</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="cmbModbusMode">
        <item>
         <property name="text">
          <string>RTU</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>TCP</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="lblSlaveID">
        <property name="text">
         <string>Slave ID</string>
        </property>
        <property name="buddy">
         <cstring>sbSlaveID</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QSpinBox" name="sbSlaveID">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>255</number>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QLabel" name="lblFunctionCode">
        <property name="text">
         <string>Function Code</string>
        </property>
        <property name="buddy">
         <cstring>cmbFunctionCode</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QComboBox" name="cmbFunctionCode"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="lblStartAddress">
        <property name="text">
         <string>Start Address</string>
        </property>
        <property name="buddy">
         <cstring>sbStartAddress</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="sbStartAddress">
        <property name="maximum">
         <number>65535</number>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="lblNoOfCoils">
        <property name="text">
         <string>Number of Coils</string>
        </property>
        <property name="buddy">
         <cstring>sbNoOfCoils</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QSpinBox" name="sbNoOfCoils">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
       </widget>
      </item>
      <item row="1" column="5">
       <widget class="QComboBox" name="cmbBase">
        <property name="currentIndex">
         <number>1</number>
        </property>
        <item>
         <property name="text">
          <string>Binary</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Decimal</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Hex</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="lblCmds">
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Commands</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayoutCmds">
      <item row="0" column="2">
       <widget class="QCheckBox" name="chkReqCycle">
        <property name="text">
         <string>Request Cyclicaly (msec)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QSpinBox" name="spInterval">
        <property name="minimum">
         <number>500</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
        <property name="singleStep">
         <number>500</number>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="btConnect">
        <property name="toolTip">
         <string>Connect</string>
        </property>
        <property name="text">
         <string>Connect</string>
        </property>
        <property name="icon">
         <iconset resource="../data/qModMaster.qrc">
          <normaloff>:/img/connect-16.png</normaloff>:/img/connect-16.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>16</width>
          <height>16</height>
         </size>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="btRequest">
        <property name="toolTip">
         <string>Tx/Rx Data</string>
        </property>
        <property name="text">
         <string>Request</string>
        </property>
        <property name="icon">
         <iconset resource="../data/qModMaster.qrc">
          <normaloff>:/img/refresh-16.png</normaloff>:/img/refresh-16.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>16</width>
          <height>16</height>
         </size>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="btAddItems">
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="toolTip">
         <string>Add Items</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../data/qModMaster.qrc">
          <normaloff>:/img/add-16.png</normaloff>:/img/add-16.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>16</width>
          <height>16</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QPushButton" name="btClear">
        <property name="toolTip">
         <string>Clear Items</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../data/qModMaster.qrc">
          <normaloff>:/img/remove-16.png</normaloff>:/img/remove-16.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>16</width>
          <height>16</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="lblRegisters">
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Registers &gt;</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTableView" name="tblRegisters">
      <attribute name="horizontalHeaderVisible">
       <bool>true</bool>
      </attribute>
      <attribute name="horizontalHeaderHighlightSections">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>707</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>Options</string>
    </property>
    <addaction name="actionSerial_RTU"/>
    <addaction name="actionTCP"/>
    <addaction name="separator"/>
    <addaction name="actionSettings"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionBus_Monitor"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuOptions"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="../data/qModMaster.qrc">
     <normaloff>:/img/close16.png</normaloff>:/img/close16.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionSerial_RTU">
   <property name="icon">
    <iconset resource="../data/qModMaster.qrc">
     <normaloff>:/img/edit-16.png</normaloff>:/img/edit-16.png</iconset>
   </property>
   <property name="text">
    <string>Modbus RTU...</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionTCP">
   <property name="icon">
    <iconset resource="../data/qModMaster.qrc">
     <normaloff>:/img/network-16.png</normaloff>:/img/network-16.png</iconset>
   </property>
   <property name="text">
    <string>Modbus TCP...</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="../data/qModMaster.qrc">
     <normaloff>:/img/info16.png</normaloff>:/img/info16.png</iconset>
   </property>
   <property name="text">
    <string>About...</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionBus_Monitor">
   <property name="icon">
    <iconset resource="../data/qModMaster.qrc">
     <normaloff>:/img/view-16.png</normaloff>:/img/view-16.png</iconset>
   </property>
   <property name="text">
    <string>Bus Monitor</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionSettings">
   <property name="icon">
    <iconset resource="../data/qModMaster.qrc">
     <normaloff>:/img/options-16.png</normaloff>:/img/options-16.png</iconset>
   </property>
   <property name="text">
    <string>Settings...</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>chkReqCycle</tabstop>
  <tabstop>spInterval</tabstop>
  <tabstop>tblRegisters</tabstop>
 </tabstops>
 <resources>
  <include location="../data/qModMaster.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
